misk/src/main/kotlin/misk/client/ClientInvocationHandler.kt

```diff
@@ -39,6 +39,7 @@ internal class ClientInvocationHandler(
   private val clientName: String,
   retrofit: Retrofit,
   okHttpTemplate: OkHttpClient,
+  clientInterceptorFactories: Provider<List<ClientInterceptors.Factory>>,
   networkInterceptorFactories: Provider<List<ClientNetworkInterceptor.Factory>>,
   applicationInterceptorFactories: Provider<List<ClientApplicationInterceptor.Factory>>,
```

Comment:
I get why you're doing this, but it doesn't seem satisfying that we have `ClientApplicationInterceptor.Factory` for retrofit and `ClientInterceptors.Factory` for okhttp.  I almost would argue we should remove the abstraction and call them `OkHttpInterceptor` and `RetrofitInterceptor`